MovingBlocks/TerasologyLauncher
# Adds gradle task to generate Web API client

### Summary
Adds gradle task `generateSwaggerCodeClient` to generate web API config using swagger specs from `api-config.json`. This can then be compiled and used to interact with the REST API of the server facade. It aims to provide core functionalities for the upcoming [Server manager](https://github.com/MovingBlocks/TerasologyLauncher/issues/427). For documentation on the REST API see [this](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/MovingBlocks/FacadeServer/develop/src/main/resources/web/swagger.json#/).

### Testing
- Go to the project root and run `gradlew generateSwaggerCodeClient`
- A new directory `web-api-client` should appear containing sources for the web API client
- Use `gradlew idea` in the project root to import the generated directory as a subproject. IntelliJ may require a restart afterwards.
